### PR TITLE
Add support for filtering out metrics for mounted ufs

### PR DESCRIPTION
**Pull request by cheng-chang**
When a ufs is unmounted, its metrics still exist in the metrics system, this PR filters out ufs metrics for the mounted ufs only.

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java [hunk 1]
```diff
@@ -839,6 +839,15 @@ public Response getWebUIWorkers() {
     }, ServerConfiguration.global());
   }
 
+  private boolean isMounted(String ufs) {
```
Comment (madanadit) [inline core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java]: nit: add javadoc

File: core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java [hunk 2]
```diff
@@ -936,14 +949,22 @@ public Response getWebUIMetrics() {
...
       // cluster per UFS read
       Map<String, String> ufsReadSizeMap = new TreeMap<>();
       for (Map.Entry<String, Gauge> entry : mr
           .getGauges((name, metric) -> name.contains(WorkerMetrics.BYTES_READ_UFS)).entrySet()) {
         alluxio.metrics.Metric metric =
             alluxio.metrics.Metric.from(entry.getKey(), (long) entry.getValue().getValue());
-        ufsReadSizeMap.put(metric.getTags().get(WorkerMetrics.TAG_UFS),
-            FormatUtils.getSizeFromBytes((long) metric.getValue()));
+        String ufs = metric.getTags().get(WorkerMetrics.TAG_UFS);
+        if (isMounted(ufs)) {
```
Comment (madanadit) [inline core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java]: @cheng-chang is it easy to add a test case?

Comment (cheng-chang) [reply core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java]: the existing test uses a lot of mocks, it's not easy to add a test for this.

File: core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java (hunk 1 above)
Comment (cheng-chang) [reply core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java]: done

File: core/server/master/src/test/java/alluxio/master/meta/AlluxioMasterRestServiceHandlerTest.java [hunk 3]
```diff
@@ -431,4 +432,19 @@ public void getWorkerInfoList() {
...
+  public void isMounted() throws Exception {
+    String s3Uri = "s3a://test/dir_1/dir-2";
+    String hdfsUri = "hdfs://test";
+
+    Map<String, MountPointInfo> mountTable = new HashMap<>();
+    mountTable.put("/s3", new MountPointInfo().setUfsUri(s3Uri));
+    when(mFileSystemMaster.getMountTable()).thenReturn(mountTable);
+
+    assertFalse(mHandler.isMounted(s3Uri));
+    assertTrue(mHandler.isMounted(MetricsSystem.escape(new AlluxioURI(s3Uri))));
```
Comment (madanadit) [inline core/server/master/src/test/java/alluxio/master/meta/AlluxioMasterRestServiceHandlerTest.java]: can you also check if this succeeds for s3Uri and hdfsUri ending with /. is mountTable key guaranteed to not end with /?

Comment (cheng-chang) [reply core/server/master/src/test/java/alluxio/master/meta/AlluxioMasterRestServiceHandlerTest.java]: the ufs uri in mount table  is guaranteed not to end with `/`

Comment (cheng-chang) [reply core/server/master/src/test/java/alluxio/master/meta/AlluxioMasterRestServiceHandlerTest.java]: to be safer, I added logic to trim the trailing slash

**Comment by cheng-chang** [comment]
jenkins, test this please

**Comment by cheng-chang** [comment]
jenkins, test this please

File: core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java [hunk 4]
```diff
@@ -839,6 +840,27 @@ public Response getWebUIWorkers() {
     }, ServerConfiguration.global());
   }
 
+  /**
+   * @param ufs the ufs uri encoded by {@link MetricsSystem#escape(AlluxioURI)}
+   * @return whether the ufs uri is a mount point
+   */
+  @VisibleForTesting
+  boolean isMounted(String ufs) {
+    if (ufs.endsWith("/")) {
```
Comment (madanadit) [inline core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java]: nit: add this to PathUtils and use AlluxioURI.SEPARATOR if a similar function does not exist?

Comment (cheng-chang) [reply core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java]: done

**Pull request merged**